app/controllers/users.ts

```diff
@@ -30,38 +30,52 @@ export function getUsers(req: Request, res: Response, next: NextFunction): void
   }
 }
 
-export async function createUser(req: Request, res: Response, next: NextFunction): Promise<void> {
-  const data = req.body;
-  const errorEmail = await checkEmail(data.email);
-  const errorPassword = checkPassword(data.password);
-  if (errorEmail) {
-    logger.error(errorEmail.message);
-    next(errorEmail);
-    return;
-  }
-  if (errorPassword) {
-    logger.error(errorPassword);
-    next(unprocessableEntity(errorPassword));
-    return;
-  }
-  const salt: number = process.env.SALT_PASSWORD ? Number(process.env.SALT_PASSWORD) : 10;
-  const password = bcrypt.hashSync(data.password, salt);
-  userService
-    .createAndSave({
-      name: data.name,
-      lastName: data.lastName,
-      email: data.email,
-      password
-    } as User)
-    .then((user: User) => {
-      res.status(HttpStatus.CREATED).send({ id: user.id });
-      logger.info(`user created with email: ${data.email}`);
-    })
-    .catch(() => {
-      const msg = 'error save user in database';
-      logger.error(msg);
-      next(databaseError(msg));
-    });
+export function createUser(role: string): (req: Request, res: Response, next: NextFunction) => Promise<void> {
```

Comment:
It would be better to create a function for admin user creation, so this one doesn't get too long